2018-06-10  John David Anglin  <danglin@gcc.gnu.org>

	[BZ #23174]
	* sysdeps/unix/sysv/linux/hppa/Makefile: xfail check-execstack.

diff --git a/sysdeps/unix/sysv/linux/hppa/Makefile b/sysdeps/unix/sysv/linux/hppa/Makefile
index 68569013fb..e1637f54f5 100644
--- a/sysdeps/unix/sysv/linux/hppa/Makefile
+++ b/sysdeps/unix/sysv/linux/hppa/Makefile
@@ -2,3 +2,10 @@
 ifeq ($(subdir),stdlib)
 gen-as-const-headers += ucontext_i.sym
 endif
+
+# Supporting non-executable stacks on HPPA requires changes to both
+# the Linux kernel and glibc. The kernel currently needs an executable
+# stack for syscall restarts and signal returns.
+ifeq ($(subdir),elf)
+test-xfail-check-execstack = yes
+endif
